dependencies.yaml

```diff
@@ -591,20 +605,43 @@ dependencies:
               - pynvjitlink>=0.0.0a0
           - matrix: {cuda: "11.*"}
             packages:
-              - cubinlinker
-              - ptxcompiler
+              - &cubinlinker_conda cubinlinker
+              - &ptxcompiler_conda ptxcompiler
       - output_types: [requirements, pyproject]
         matrices:
-          - matrix: {cuda: "12.*"}
+          - matrix:
+              cuda: "12.*"
+              cuda_suffixed: "true"
             packages:
               - rmm-cu12==24.8.*,>=0.0.0a0
-              - pynvjitlink-cu12>=0.0.0a0
-          - matrix: {cuda: "11.*"}
+              - &pynvjitlink_cu12 pynvjitlink-cu12>=0.0.0a0
+          - matrix:
+              cuda: "12.*"
+              cuda_suffixed: "false"
+            packages:
+              - *rmm_conda
+              # pynvjitlink still has a '-cu12' suffix here because:
+              #
+              #   * it's suffixed when built from source in DLFW builds
+              #   * it needs to be 'pip install'-ed from package repositories
+              #     in RAPIDS devcontainer builds
+              #
+              - *pynvjitlink_cu12
```

Comment:
```suggestion
              - pynvjitlink>=0.0.0a0
```

Reply:
yup perfect, thanks for double-checking my understanding there!

Updated in https://github.com/rapidsai/cudf/pull/16183/commits/035213e88c365a318af2062d761994c144f4d186